src/linked_hash_map.rs

```diff
@@ -1677,6 +1749,173 @@ impl<'a, K, V> Drop for Drain<'a, K, V> {
+    #[inline]
+    pub fn peek_prev(&mut self) -> Option<(&K, &mut V)> {
+        unsafe {
+            let at = (*self.cur).links.value.prev;
+            self.peek(|| at)
+        }
+    }
+
+    // Retrieves the element specified by the at closure function without advancing current
+    // position to it.
+    #[inline]
+    fn peek(&mut self, at: impl FnOnce() -> NonNull<Node<K, V>>) -> Option<(&K, &mut V)> {
+        if let Some(values) = self.values {
+            unsafe {
+                let node = at().as_ptr();
+                if node == values.as_ptr() {
+                    None
+                } else {
+                    let entry = (*node).entry_mut();
+                    Some((&entry.0, &mut entry.1))
+                }
+            }
+        } else {
+            None
+        }
+    }
+
+    /// Updates the pointer to the current element to the next element in the
+    /// list (that is, moving towards the end).
+    #[inline]
+    pub fn move_next(&mut self) {
+        let at = unsafe { (*self.cur).links.value.next };
+        self.muv(|| at);
+    }
+
+    /// Updates the pointer to the current element to the previous element in the
+    /// list (that is, moving towards the front).
+    #[inline]
+    pub fn move_prev(&mut self) {
+        let at = unsafe { (*self.cur).links.value.prev };
+        self.muv(|| at);
+    }
+
+    /// Positions the cursor at the element associated with the specified key. Returns `true`
+    /// if the element exists and the operation succeeds, or `false` if the element does not
+    /// exist.
+    #[inline]
+    pub fn move_at(&mut self, key: &K) -> bool
+    where
+        K: Eq + Hash,
+        S: BuildHasher,
+    {
+        unsafe {
+            let hash = hash_key(self.hash_builder, key);
+            let i_entry = self
+                .table
+                .find_entry(hash, |o| (*o).as_ref().key_ref().eq(key));
+
+            match i_entry {
+                Ok(occupied) => {
+                    let at = *occupied.get();
+                    self.muv(|| at);
+                    true
+                }
+                Err(_) => false,
+            }
+        }
+    }
+
+    // Updates the pointer to the current element to the one returned by the at closure function.
+    #[inline]
+    fn muv(&mut self, at: impl FnOnce() -> NonNull<Node<K, V>>) {
+        self.cur = at().as_ptr();
+    }
+
+    /// Inserts the provided key and value before the current item. It checks if an entry
+    /// with the given key exists and, if so, replaces its value with the provided `1`
+    /// parameter from the given tuple. The key is not updated; this matters for types that
+    /// can be `==` without being identical.
+    ///
+    /// If the entry doesn't exist, it creates a new one. If a value has been updated, the
+    /// function returns the *old* value wrapped with `Some`  and `None` otherwise.
+    #[inline]
+    pub fn insert_before(&mut self, tuple: (K, V)) -> Option<V>
```

Comment:
All of the other methods that insert entries always take the key and value as separate parameters and we should match it here, and the same thing for `insert_after`.

Reply:
Sure, I'll make that change. Thanks for bringing it to my attention!